script.js

```diff
@@ -1,4 +1,130 @@
+var numOne = 0;
+var numTwo = 0;
+var numThree = 0;
+var numFour = 0;
+var numFive = 0;
+var numSix = 0;
+var prevRoll = 0;
+var numOfWin = 0;
+var numOfLoss = 0;
+var numOfGame = 0;
+var points = 0;
+var result = "";
+var lossRate = 0;
+var winRate = 0;
+var currentGameMode = "waiting for user name";
+var userName = "";
+var userGuess = 0;
+
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+var diceGame = function (userName, userGuess) {
+  var message = "";
+  var randomDiceNumber = rollDice();
+  numOfGame = numOfGame + 1;
+  // determine if this is the first roll
+  if (prevRoll == 0) {
+    message = `Hi ${userName}! This is your first roll! `;
+    console.log(message);
+  } else {
+    message = `Hi ${userName}! Your last roll was ${prevRoll}.`;
+  }
+  // determine it is a win/loss
+  if (randomDiceNumber == userGuess) {
+    numOfWin = numOfWin + 1;
+    result = "win";
+  } else {
+    numOfLoss = numOfLoss + 1;
+    result = "lose";
+  }
+  winRate = ((numOfWin / numOfGame) * 100).toFixed(0) + `%`;
+  lossRate = ((numOfLoss / numOfGame) * 100).toFixed(0) + `%`;
+  message += `You just rolled a ${randomDiceNumber}. You guessed ${userGuess}. You ${result}! <br>
+  You win ${numOfWin} out of ${numOfGame}. Win rate = ${winRate}.<br>
+  You lose ${numOfLoss} out of ${numOfGame}. Loss rate = ${lossRate}.<br>`;
+  console.log(message);
+  prevRoll = randomDiceNumber;
+  // determine which is the most rolled number
```

Comment:
you can consider putting your program logic that determines the most rolled number (lines 52 - 97) into a helper function